src/engine/tests/Test_SigmoidConstraint.h

```diff
@@ -0,0 +1,322 @@
+        auto it = participatingVariables.begin();
+        TS_ASSERT_EQUALS( *it, b );
+        ++it;
+        TS_ASSERT_EQUALS( *it, f );
+
+        TS_ASSERT( sigmoid.participatingVariable( b ) );
+        TS_ASSERT( sigmoid.participatingVariable( f ) );
+        TS_ASSERT( !sigmoid.participatingVariable( new_b ) );
+        TS_ASSERT( !sigmoid.participatingVariable( new_f ) );
+
+        sigmoid.notifyLowerBound( b, -1.0 );
+        sigmoid.notifyLowerBound( f, 0.268 );
+        sigmoid.notifyUpperBound( b, 1.0 );
+        sigmoid.notifyUpperBound( f, 0.731 );
+
+
+        // update variable index
+        sigmoid.updateVariableIndex( b, new_b );
+        sigmoid.updateVariableIndex( f, new_f );
+
+        TS_ASSERT( !sigmoid.participatingVariable( b ) );
+        TS_ASSERT( !sigmoid.participatingVariable( f ) );
+        TS_ASSERT_EQUALS( sigmoid.getB(), new_b );
+        TS_ASSERT_EQUALS( sigmoid.getF(), new_f );
+        TS_ASSERT( sigmoid.participatingVariable( new_b ) );
+        TS_ASSERT( sigmoid.participatingVariable( new_f ) );
+
+        List<Tightening> tightenings;
+        sigmoid.getEntailedTightenings( tightenings );
+
+        auto it2 = tightenings.begin();
+        TS_ASSERT_EQUALS( it2->_type, Tightening::LB );
+        TS_ASSERT_EQUALS( it2->_variable, new_b );
+        TS_ASSERT_EQUALS( it2->_value, -1.0 );
+
+        ++it2;
+        TS_ASSERT_EQUALS( it2->_type, Tightening::LB );
+        TS_ASSERT_EQUALS( it2->_variable, new_f );
+        TS_ASSERT_EQUALS( it2->_value, 0.268 );
+
+        ++it2;
+        TS_ASSERT_EQUALS( it2->_type, Tightening::UB );
+        TS_ASSERT_EQUALS( it2->_variable, new_b );
+        TS_ASSERT_EQUALS( it2->_value, 1.0 );
+
+        ++it2;
+        TS_ASSERT_EQUALS( it2->_type, Tightening::UB );
+        TS_ASSERT_EQUALS( it2->_variable, new_f );
+        TS_ASSERT_EQUALS( it2->_value, 0.731 );
+    }
+
+    void test_sigmoid_serialize()
+    {
+        unsigned b = 0;
+        unsigned f = 1;
+
+        SigmoidConstraint sigmoid( b, f );
+    
+        String serializedString = sigmoid.serializeToString();
+
+        SigmoidConstraint serializedSigmooid( serializedString.ascii() );
+
+        TS_ASSERT_EQUALS( serializedSigmooid.getB(), b );
+        TS_ASSERT_EQUALS( serializedSigmooid.getF(), f );
+    }
+
+    void test_sigmoid_restore()
+    {
+        unsigned b_1 = 0;
+        unsigned f_1 = 1;
+        unsigned b_2 = 2;
+        unsigned f_2 = 3;
+
+        SigmoidConstraint sigmoid1( b_1, f_1 );
+        TS_ASSERT_EQUALS( sigmoid1.getB(), b_1 );
+        TS_ASSERT_EQUALS( sigmoid1.getF(), f_1 );
+
+        SigmoidConstraint sigmoid2( b_2, f_2 );
+        TS_ASSERT_EQUALS( sigmoid2.getB(), b_2 );
+        TS_ASSERT_EQUALS( sigmoid2.getF(), f_2 );
+        
+        // restore
+        sigmoid1.restoreState( &sigmoid2 );
+        TS_ASSERT_EQUALS( sigmoid1.getB(), b_2 );
+        TS_ASSERT_EQUALS( sigmoid1.getF(), f_2 );        
+    }
```

Comment:
Maybe also add [test_register_as_watcher() ](https://github.com/NeuralNetworkVerification/Marabou/blob/22c4998faa5549d06c54235e5a8925906cf6a643/src/engine/tests/Test_ReluConstraint.h#L352) 
and 
[test_sigmoid_entailed_tightening()](https://github.com/NeuralNetworkVerification/Marabou/blob/22c4998faa5549d06c54235e5a8925906cf6a643/src/engine/tests/Test_ReluConstraint.h#L643)

Reply:
Added.